compact/nodes_test.go

```diff
@@ -118,3 +118,30 @@ func refRangeNodes(root NodeID, begin, end uint64) []NodeID {
 		refRangeNodes(NewNodeID(root.Level-1, root.Index*2), begin, end),
 		refRangeNodes(NewNodeID(root.Level-1, root.Index*2+1), begin, end)...)
 }
+
+// Test that RangeNodes returns a slice of nodes with contiguous coverage.
+// https://github.com/transparency-dev/merkle/blob/main/docs/compact_ranges.md#definition
+func FuzzRangeNodes(f *testing.F) {
+	f.Fuzz(func(t *testing.T, begin, end uint64) {
+		if begin > end {
+			return
+		}
+		t.Logf("begin=%d, end=%d", begin, end)
+		ids := make([]NodeID, 0)
+		nodes := RangeNodes(begin, end, ids)
```

Comment:
Can ding the `ids := ...` line and change the call to `RangeNodes` to pass in `nil` as the last param (it acts like `append` would in that regard)